express-post-json/index.js

```diff
@@ -0,0 +1,27 @@
+const express = require('express');
+const app = express();
+let nextID = 1;
+const grades = {};
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('Listening on port 3000!');
+});
+
+app.get('/api/grades', function (req, res, next) {
+  const array = [];
+  for (const i in grades) {
+    array.push(grades[i]);
+  }
+  res.json(array);
+});
+
+app.use(express.json());
+
+app.post('/api/grades', function (req, res, next) {
+  grades.nextID = req.body;
+  grades.nextID.id = nextID;
+  res.status(201);
+  res.send(grades.nextID);
+  nextID++;
```

Comment:
Although this will work here we don't want to do any work after sending a response to the client. Sending a response does NOT stop code execution but it is still best practice to perform all actions before sending the response. Sending the response should always be the last thing to happen.